lego/apps/email/tasks.py

```diff
@@ -0,0 +1,23 @@
+from structlog import get_logger
+
+from lego import celery_app
+from lego.apps.email.notifications import WeeklyNotification
+from lego.apps.users.models import AbakusGroup
+from lego.utils.tasks import AbakusTask
+
+log = get_logger()
+
+
+@celery_app.task(serializer="json", bind=True, base=AbakusTask)
+def send_weekly_email(self, logger_context=None):
+    # Set to just PR and Webkom for testing purposes
+    all_users = set(
+        AbakusGroup.objects.get(name="Webkom").restricted_lookup()[0]
+        + AbakusGroup.objects.get(name="PR").restricted_lookup()[0]
+    )
+
+    self.setup_logger(logger_context)
+
+    for user in all_users:
+        notification = WeeklyNotification(user)
+        notification.notify()
```

Comment:
This is, sadly, not the way to do this :/

For prod, we cant really do this at all. Considering you are using the user for permission context I understand the reasoning, but we _cannot_ send an individual email to ALL users.

You will need to make the email tailored for the "average" user and just make sure that the permissions are based on the group you are sending too (e.g. `Abakus`), and batch send the email to all users in that group. For django, this is done using https://docs.djangoproject.com/en/4.0/topics/email/#send-mass-mail.

Reply:
It might be fine for the testing though, seen as it is limited to the groups above for now, but I think it's a good idea to test with the "production ready" code when we're first testing.

Reply:
I think you have misunderstood https://docs.djangoproject.com/en/4.0/topics/email/#send-mass-mail. It is not made for sending the same email to a mass of users, but sending out _different_ emails at the same time. I agree that what I have done is not working, but we do not **not** have to send the same email to all users. Creating an email per user, then using a single call of `send_mass_email` should suffice.

Reply:
Yep, I did not pay attention there. You are of course correct in that we don't need to send the same email :)